### PR TITLE
add TripletAngularMarginLoss_XBM and refine XBM queue

### PR DESCRIPTION
**Pull request by HydrogenSulfate**
主要功能(from PR #2359)：

1. 添加基于`CrossBatchMemory`特征队列模块的`TripletAngularMarginLoss_XBM`
2. 完善`CrossBatchMemory`的初始化机制，将队列相关变量注册为buffer，以保证XBM模块在加载checkpoints模型时的正确性

其他：
1. export方法支持http/https格式的`Global.pretrained_model`字段

### CODE REVIEW

File: ppcls/engine/engine.py [hunk 1]
```diff
@@ -548,7 +548,10 @@ def export(self):
             "use_multilabel",
             False) or "ATTRMetric" in self.config["Metric"]["Eval"][0]
         model = ExportModel(self.config["Arch"], self.model, use_multilabel)
-        if self.config["Global"]["pretrained_model"] is not None:
+        if self.config["Global"]["pretrained_model"].startswith("http"):
+            load_dygraph_pretrain_from_url(
+                model.base_model, self.config["Global"]["pretrained_model"])
+        else:
```
Comment (RainFrost1) [inline ppcls/engine/engine.py]: 我记得好像支持过http的形式。另外这个修改了之后，为None的情况应该也得判断一下吧？

File: ppcls/loss/tripletangularmarginloss.py [hunk 2]
```diff
@@ -113,3 +114,128 @@ def forward(self, input, target):
...
+
+            absolut_loss_an = dist_an - self.an_value
+            absolut_loss_an = paddle.where(absolut_loss_an > 0,
+                                           absolut_loss_an,
+                                           paddle.ones_like(absolut_loss_an))
+
+            loss = (absolut_loss_an.mean() + absolut_loss_ap.mean()
+                    ) * self.absolute_loss_weight + loss.mean()
+
+        return loss
```
Comment (RainFrost1) [inline ppcls/loss/tripletangularmarginloss.py:241]: 返回的loss应该是个字典形式吧？

File: ppcls/engine/engine.py (hunk 1 above)
Comment (HydrogenSulfate) [reply ppcls/engine/engine.py]: > 我记得好像支持过http的形式。另外这个修改了之后，为None的情况应该也得判断一下吧？

之前适配的是`Engine.__init__`里，加载pretrainde_model，`Engine.export`没有适配

File: ppcls/loss/tripletangularmarginloss.py (hunk 2 above)
Comment (HydrogenSulfate) [reply ppcls/loss/tripletangularmarginloss.py:241]: > 返回的loss应该是个字典形式吧？

这个是内部函数，单纯计算loss值，字典形式返回在159行的def forward(self, input, target):里

**Review by RainFrost1 — APPROVED** [review]
